rocketpy/motors/tank_geometry.py

```diff
@@ -369,28 +369,46 @@ def __init__(self, radius, height, spherical_caps=False, geometry_dict=dict()):
         height : float
             Height of the cylindrical tank, in meters.
         spherical_caps : bool, optional
-            If True, the tank will have spherical caps. The default is False.
+            If True, the tank will have spherical caps at the top and bottom
+            with the same radius as the cylindrical part. If False, the tank
+            will have flat caps at the top and bottom. Defaults to False.
         geometry_dict : dict, optional
             Dictionary containing the geometry of the tank. See TankGeometry.
         """
         super().__init__(geometry_dict)
+        self.height = height
         self.has_caps = False
-        self.add_geometry((-height / 2, height / 2), radius)
-        self.add_spherical_caps() if spherical_caps else None
+        if spherical_caps:
+            self.add_geometry((-height / 2 + radius, height / 2 - radius), radius)
+            self.add_spherical_caps()
+        else:
+            self.add_geometry((-height / 2, height / 2), radius)
 
     def add_spherical_caps(self):
         """
         Adds spherical caps to the tank. The caps are added at the bottom
-        and at the top of the tank. If the tank already has caps, it raises a
-        ValueError.
-        """
+        and at the top of the tank with the same radius as the cylindrical
+        part. The height is not modified, meaning that the total volume of
+        the tank will decrease.
+        """
+        # raise warning
+        print(
+            "Warning: Adding spherical caps to the tank will not modify the "
+            + "height of the tank."
+        )
```

Comment:
A would rather the warning was a bit more clear on what is the method outcome, for example:

```suggestion
        print(
            "Warning: Adding spherical caps to the tank will not modify the "
            + f"total height of the tank {self.height}."
            + f"Its cylindrical portion height will be reduced to {self.height - 2*self.radius}."
        )
```